CRM/Report/Form.php

```diff
@@ -4172,6 +4173,41 @@ public function alterDisplayAddressFields(&$row, &$rows, &$rowNum, $baseUrl, $ur
   }
 
   /**
+   * Do AlterDisplay processing on Address Fields.
+   *
+   * @param array $row
+   * @param array $rows
+   * @param int $rowNum
+   *
+   * @return bool
+   */
+  public function alterDisplayContactFields(&$row, &$rows, &$rowNum) {
+    $entryFound = FALSE;
+    if (array_key_exists('civicrm_contact_prefix_id', $row)) {
+      $prefixes = CRM_Contact_BAO_Contact::buildOptions('prefix_id');
+      if ($value = $row['civicrm_contact_prefix_id']) {
+        $rows[$rowNum]['civicrm_contact_prefix_id'] = $prefixes[$rows[$rowNum]['civicrm_contact_prefix_id']];
+      }
+      $entryFound = TRUE;
+    }
+    if (array_key_exists('civicrm_contact_suffix_id', $row)) {
+      $suffixes = CRM_Contact_BAO_Contact::buildOptions('suffix_id');
+      if ($value = $row['civicrm_contact_suffix_id']) {
+        $rows[$rowNum]['civicrm_contact_suffix_id'] = $suffixes[$rows[$rowNum]['civicrm_contact_suffix_id']];
+      }
+      $entryFound = TRUE;
+    }
+    if (array_key_exists('civicrm_contact_gender_id', $row)) {
+      $genders = CRM_Contact_BAO_Contact::buildOptions('gender_id');
```

Comment:
probably - I didn't know about that when I did this. If I need to re-visit this PR I'll change it